BaikalDataGenerator/Source/data_generator.h

```diff
@@ -22,35 +22,58 @@ THE SOFTWARE.
 
 #pragma once
 
-#include "utils.h"
-#include "input_info.h"
-#include "filesystem.h"
+#include "Rpr/RadeonProRender.h"
 
-#include <vector>
+#ifdef __cplusplus
+extern "C" {
+#endif
 
-using CameraIterator = std::vector<CameraInfo>::const_iterator;
-using LightsIterator = std::vector<LightInfo>::const_iterator;
-using SppIterator = std::vector<int>::const_iterator;
-
-class ConfigLoader
+struct DataGeneratorParams
 {
-public:
-    explicit ConfigLoader(const DGenConfig& config);
+    char const* output_dir;
+
+    rpr_scene scene;
+    char const* scene_name;
+
+    rpr_camera* cameras;
+    unsigned cameras_num;
+    int cameras_start_output_idx;
+
+    rpr_light* lights;
+    unsigned lights_num;
 
-    std::vector<CameraInfo> CamStates() const;
-    std::vector<LightInfo> Lights() const;
-    const std::filesystem::path& LightsDir() const;
-    std::vector<size_t> Spp() const;
+    unsigned const* spp;
+    unsigned spp_num;
 
-    private:
+    unsigned width;
+    unsigned height;
 
-    void ValidateConfig(const DGenConfig& config) const;
+    unsigned bounces_num;
+    unsigned device_idx;
+
+    unsigned gamma_correction;
+
+    void (*progress_callback)(int /* camera_idx */);
```

Comment:
lets change signature to:
void (*progress_callback)(unsigned done, unsigned total);